server/neptune/workflows/internal/activities/terraform.go

```diff
@@ -59,28 +66,49 @@ func (t *terraformActivities) TerraformInit(ctx context.Context, request Terrafo
 	}
 
 	ch := t.TerraformClient.RunCommand(ctx, request.JobID, request.Path, cmd.Build(), request.Envs, tfVersion)
-	var lines []string
-	for line := range ch {
-		if line.Err != nil {
-			err = errors.Wrap(line.Err, "executing command")
-			break
-		}
-		lines = append(lines, line.Line)
+	_, err = t.readTerraformOutput(ch)
+	if err != nil {
+		return TerraformInitResponse{}, errors.Wrap(err, "processing command output")
 	}
-	output := strings.Join(lines, "\n")
-
-	// sanitize output by stripping out any ansi characters.
-	output = ansi.Strip(output)
 	return TerraformInitResponse{}, nil
 }
 
 // Terraform Plan
-
 type TerraformPlanRequest struct {
+	Step      job.Step
+	Envs      map[string]string
+	JobID     string
+	TfVersion string
+	Path      string
 }
 
-func (t *terraformActivities) TerraformPlan(ctx context.Context, request TerraformPlanRequest) error {
-	return nil
+type TerraformPlanResponse struct {
+	PlanFile string
+	Output   string
+}
+
+func (t *terraformActivities) TerraformPlan(ctx context.Context, request TerraformPlanRequest) (TerraformPlanResponse, error) {
+	tfVersion, err := t.resolveVersion(request.TfVersion)
+	if err != nil {
+		return TerraformPlanResponse{}, err
+	}
+	planFile := filepath.Join(request.Path, PlanOutputFile)
+	cmd, err := terraform.NewCommandArguments(
+		terraform.Plan,
+		[]string{DisableInputArg, fmt.Sprintf("%s=%s", RefreshArg, "true"), fmt.Sprintf("%s=%s", OutArg, planFile)},
```

Comment:
nit: should we just set the args (similar to `DisableInputArg`) as constant so that we don't have to build the string here?